Add RabbitMQ Component

eerhardt opened the pull request:
Fix #231

eerhardt committed:
Initial RabbitMQ component

eerhardt committed:
Update README and ConfigurationSchema

eerhardt committed:
Add IsAspireHost to AppHost projects (#531)

tlmii committed:
Fix structured log custom color theme switching (#532)

baronfel committed:
explicitly support C# for aspire codegen (#472)

* explicitly support C# for aspire codegen

* add FileWrites and language-support check

* slight fixes to ServiceNameOverride templating generation

* escape dashes in project names

* only write service metadata files when the content changes

danegsta committed:
Add initial check for missing Aspire.Hosting package (#512)

* Add initial check for missing Aspire.Hosting package

* Update to be consistent with other proposed build warnings

* AppHost instead of Host

danegsta committed:
Prepare to make Aspire install by default on upgrades for web workload in VS (#538)

tlmii committed:
Text alignment with ellipsis (#536)

tlmii committed:
Metrics page alignment (#537)

eerhardt committed:
Complete RabbitMQ

baronfel committed:
Provide an explicit target to generate the Aspire manifest (#534)

mitchdenny reviewed (approved):
Lgtm. Same comments around dead code as David. Also make sure that you run the manifest generation to produce a fresh aspire-manifest.json and be happy with the output should someone try to deploy this to ACA.

davidfowl committed:
Fix one place where service discovery isn't AOT compatible (#540)

eerhardt committed:
Clean up commented out code.

davidfowl committed:
Updated otel versions to latest (#542)

JamesNK committed:
Add log level select (#541)

tlmii committed:
Display Container ID (#544)

eerhardt committed:
Enable ConfigurationBinder source gen on components by default (#535)

* Enable ConfigurationBinder source gen on all components.

Also enable IsAotCompatible, so we get warnings when using non-AOT compatible code.

* Make the Source Generator actually work by referencing Microsoft.Extensions.Configuration.Binder, which brings in the latest version that contains the source generator.

* Revert Azure components due to Pdb2Pdb error for extern alias.

eerhardt committed:
Finish up configuration for RabbitMQ.
